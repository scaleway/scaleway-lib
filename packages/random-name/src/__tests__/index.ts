import randomName from '..'

describe('randomName', () => {
  it('should return a random name separated by a dash', () => {
    expect(randomName().split('-').length).toBe(2)
  })

  it('should be able to customize prefix', () => {
    const name = randomName('prefix')
    expect(name.split('-')[0]).toBe('prefix')
    expect(name.split('-').length).toBe(3)
  })

  it('should be able to customize separator', () => {
    expect(randomName('', '/').split('/').length).toBe(2)
  })

  it('should be able to customize prefix and separator', () => {
    const name = randomName('prefix', '!')
    expect(name.split('!')[0]).toBe('prefix')
    expect(name.split('!').length).toBe(3)
  })

  it('should never have boring-wozniak', () => {
    const names = Array.from(Array(1000000), () => randomName())
    expect(names).not.toEqual(
      expect.arrayContaining([expect.stringMatching('boring-wozniak')]),
    )
  })
})
