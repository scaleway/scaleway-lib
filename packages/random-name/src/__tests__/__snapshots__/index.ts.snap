// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`randomName useRandomName useTranslation should not be defined without I18nProvider 1`] = `"gracious-hermann"`;
